akka-stream-tests/src/test/scala/akka/stream/scaladsl/ActorRefSourceSpec.scala

```diff
@@ -129,6 +130,22 @@ class ActorRefSourceSpec extends StreamSpec {
       s.expectComplete()
     }
 
+    "complete and materialize the stream after receiving Status.Success" in assertAllStagesStopped {
+      val (ref, done) = {
+        Source.actorRef(3, OverflowStrategy.dropBuffer).toMat(Sink.ignore)(Keep.both).run()
+      }
+      ref ! Status.Success("ok")
+      done.futureValue should be(Done)
+    }
+
+    "complete and materialize the stream after receiving PoisonPill" in assertAllStagesStopped {
+      val (ref, done) = {
+        Source.actorRef(3, OverflowStrategy.dropBuffer).toMat(Sink.ignore)(Keep.both).run()
+      }
+      ref ! PoisonPill
```

Comment:
This one is trickier because the actor is stopped by the PoisonPill before the subscription arrives so the aroundPostStop can't do much here https://github.com/akka/akka/blob/master/akka-stream/src/main/scala/akka/stream/actor/ActorPublisher.scala#L382

The ActorRefSourceActor is implemented with the deprecated ActorPublisher so I think we should try to rewrite that as a GraphStage instead. I suggest fixing the Success case in this PR and leave this test as pending. Then we'll create a new ticket for the rewrite.